File: Arrays/Solutions/Minimum_Pair_Removal_to_Sort_Array_I.py
class Solution:
    def minimumPairRemoval(self, nums: List[int]) -> int:
        def is_non_decreasing(arr: List[int]) -> bool:
            for i in range(1, len(arr)):
                if arr[i] < arr[i - 1]:
                    return False
            return True

        operations = 0

        while not is_non_decreasing(nums):
            min_sum = float('inf')
            idx = 0

            # Find leftmost adjacent pair with minimum sum
            for i in range(len(nums) - 1):
                s = nums[i] + nums[i + 1]
                if s < min_sum:
                    min_sum = s
                    idx = i

            # Replace the pair with their sum
            nums[idx] = min_sum
            nums.pop(idx + 1)
            operations += 1

        return operations

File: Arrays/Solutions/Minimum_Pair_Removal_to_Sort_Array_I.cpp
class Solution {
public:
    int minimumPairRemoval(vector<int>& nums) {
        int operations = 0;

        auto isNonDecreasing = [&](const vector<int>& arr) {
            for (int i = 1; i < arr.size(); i++) {
                if (arr[i] < arr[i - 1]) return false;
            }
            return true;
        };

        while (!isNonDecreasing(nums)) {
            int minSum = INT_MAX;
            int idx = 0;

            // Find leftmost adjacent pair with minimum sum
            for (int i = 0; i + 1 < nums.size(); i++) {
                int s = nums[i] + nums[i + 1];
                if (s < minSum) {
                    minSum = s;
                    idx = i;
                }
            }

            // Replace the pair with their sum
            nums[idx] = minSum;
            nums.erase(nums.begin() + idx + 1);
            operations++;
        }

        return operations;
    }
};
